pkg/version/generate/release_generate.go

```diff
@@ -0,0 +1,107 @@
+// +build release
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/blang/semver"
+	. "github.com/dave/jennifer/jen"
+	"github.com/pkg/errors"
+
+	"github.com/weaveworks/eksctl/pkg/version"
+)
+
+const versionFilename = "pkg/version/release.go"
+const defaultPreReleaseId = "dev"
+const defaultReleaseCandidate = "rc.0"
+
+func main() {
+	if len(os.Args) < 2 {
+		log.Fatal("missing argument")
+		return
+	}
+
+	command := os.Args[1]
+
+	var newVersion, newPreRelease string
+	switch command {
+	case "release":
+		newVersion, newPreRelease = prepareRelease()
+	case "release-candidate":
+		newVersion, newPreRelease = prepareReleaseCandidate()
+	case "development":
+		newVersion, newPreRelease = nextDevelopmentIteration()
+	case "print-version":
+		// Print simplified version X.Y.Z
+		fmt.Println(version.Version)
+		return
+	default:
+		log.Fatalf("unknown option %q. Expected 'release', 'release-candidate', 'development' or 'print-version'", command)
+		return
+	}
+
+	if err := writeVersionToFile(newVersion, newPreRelease, versionFilename); err != nil {
+		log.Fatalf("unable to write file: %s", err.Error())
+	}
+
+	version.Version = newVersion
+	version.PreReleaseID = newPreRelease
+	fmt.Println(version.GetVersion())
+
+	return
+}
+
+func prepareRelease() (string, string) {
+	return version.Version, ""
+}
+
+func prepareReleaseCandidate() (string, string) {
+	if strings.HasPrefix(version.PreReleaseID, "rc.") {
+		// Next RC
+		rcNumber, err := strconv.Atoi(strings.TrimPrefix(version.PreReleaseID, "rc."))
+		if err != nil {
+			log.Fatalf("cannot parse rc version from pre-release id %s", version.PreReleaseID)
+		}
+		newRC :=  rcNumber + 1
+		return version.Version, fmt.Sprintf("rc.%d", newRC)
+	}
+	return version.Version, defaultReleaseCandidate
+}
+
+func nextDevelopmentIteration() (string, string) {
+	ver, err := semver.Make(version.Version)
```

Comment:
There's a variant of `semver.Make` that panics if the version is invalid. That might be more suitable here as we do not expect `version.Version` to ever be invalid.

```suggestion
	ver := semver.MustParse(version.Version)
```

Reply:
Good idea